source/reference/modules/studio/configuration/dependency-resolver.rst

```diff
@@ -39,3 +39,21 @@ Here's a sample Dependency Resolver Configuration file (click on the triangle on
 |
 |
 
+-------------------------------
+Soft Dependencies Configuration
+-------------------------------
+.. version_tag::
+    :label: Since
+    :version: 4.2.0
+
+Soft dependencies are referenced items that are in a modified state and are not a hard dependency. When calculating soft
+dependencies, CrafterCMS follows transitive dependencies using recursive common table expressions (recursive CTE) in
```

Comment:
CrafterCMS follows dependencies recursively. [remove the algorithm and DB reference]